### PR TITLE
Ignore more exceptions in stubtest

### PR DESCRIPTION
**Pull request by JelleZijlstra**
See python/typeshed#6801.

### CODE REVIEW

File: mypy/stubtest.py [hunk 1]
```diff
@@ -269,11 +269,13 @@ def verify_typeinfo(
...
         stub_to_verify = next((t.names[entry].node for t in stub.mro if entry in t.names), MISSING)
         assert stub_to_verify is not None
-        yield from verify(
-            stub_to_verify,
-            getattr(runtime, mangled_entry, MISSING),
-            object_path + [entry],
-        )
+        try:
+            runtime_attr = getattr(runtime, mangled_entry)
+        except Exception:
```
Comment (sobolevn) [inline mypy/stubtest.py:274]: Do we need to log it somehow?

Comment (JelleZijlstra) [reply mypy/stubtest.py:274]: Not sure there's a good framework for that. Perhaps we can add a constant similar to MISSING, or fall back to a runtime value of Any if a non-AttriibuteError exception is thrown.

**Review by hauntsaninja — COMMENTED** [review]
This looks good, although it might make it possible for `verify` to be called with `verify(MISSING, MISSING, ...)`, which gives a not ideal error:
```
In [4]: print(list(verify(MISSING, MISSING, ["asdf", "fdsa"]))[0].get_description())
error: asdf.fdsa is not present in stub
Stub:
MISSING
Runtime:
MISSING

**Comment by JelleZijlstra** [comment]
I don't think that would be new from this change, since it already did `getattr(..., ..., MISSING)`.

**Comment by hauntsaninja** [comment]
It is new. The way we source entries, they're guaranteed to exist either in the stub or at runtime. Now if we have something that isn't in the stubs, shows up in `vars` but raises an error when you getattr it, we'll end up with MISSING, MISSING. It's niche and I thought stubtest would choke, but it looks like it'd just ends up giving a slightly confusing message, which is fine.

Actually, we should not be calling `verify` if getattr fails with an unexpected Exception. So something like:
```
try:
    runtime_attr = getattr(runtime, mangled_entry, MISSING)
except Exception:
    pass
else:
    yield from verify(stub_to_verify, runtime_attr, object_path + [entry])
```

**Comment by JelleZijlstra** [comment]
Thanks, I applied that suggestion.

**Comment by hauntsaninja** [comment]
Looks like githubstatus.com just acknowledged issues with Github Actions (I ran into issues on another repo as well)

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**